rust/flatbuffers/src/builder.rs

```diff
@@ -30,6 +31,80 @@ use crate::vector::Vector;
 use crate::vtable::{field_index_to_field_offset, VTable};
 use crate::vtable_writer::VTableWriter;
 
+/// Trait to implement custom allocation strategies for [`FlatBufferBuilder`].
+///
+/// An implementation can be used with [`FlatBufferBuilder::new_in`], enabling a custom allocation
+/// strategy for the [`FlatBufferBuilder`].
+pub trait Allocator: DerefMut<Target = [u8]> {
```

Comment:
I think `Allocator` should be considered unsafe since internal unsafe code assumes the current implementation and a dumb implementation of `grow_downwards` could cause UB.